RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -249,22 +249,59 @@ namespace SDL {
                                                        unsigned int& firstMDIndex,
                                                        unsigned int& secondMDIndex,
                                                        unsigned int& thirdMDIndex,
-                                                       float& residual) {
-    //get the rt and z
-    const float& r1 = mdsInGPU.anchorRt[firstMDIndex];
-    const float& r2 = mdsInGPU.anchorRt[secondMDIndex];
-    const float& r3 = mdsInGPU.anchorRt[thirdMDIndex];
-
-    const float& z1 = mdsInGPU.anchorZ[firstMDIndex];
-    const float& z2 = mdsInGPU.anchorZ[secondMDIndex];
-    const float& z3 = mdsInGPU.anchorZ[thirdMDIndex];
+                                                       float& residual,
+                                                       float& circleRadius,
+                                                       float& circleCenterX, 
+                                                       float& circleCenterY) {
 
     //following Philip's layer number prescription
     const int layer1 = modulesInGPU.sdlLayers[innerInnerLowerModuleIndex];
     const int layer2 = modulesInGPU.sdlLayers[middleLowerModuleIndex];
     const int layer3 = modulesInGPU.sdlLayers[outerOuterLowerModuleIndex];
 
+    //get the rt and z
+    const float& r1 = mdsInGPU.anchorRt[firstMDIndex]/100; // rt is stored in the unit of cm, in the calculation below we want to use the meter unit
```

Comment:
why references? this looks confusing. Please just use values

Reply:
can the math be converted to the standard CMS length units, cm?
this continuing cm-m conversion is confusing

Reply:
I have changed into cm units

Reply:
Thank you for your comments Slava! Sorry that I didn't make it clear previously, this is pushing to our T3 rz chi2 development branch, and could have some issues or imperfections. We will make a PR to the central batch branch after we made progress. And I suggest you can wait and make comments there for once to avoid wasting your too much time on reviewing each of our piece of work. We will be clear that next time we mark in the titles : To RZ-chi2 branch, to avoid confusion. Thank you again!